DSharpPlus.Core/Rest/PollyRetryPolicy.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+using Polly;
+using Polly.Retry;
+
+namespace DSharpPlus.Core.Rest
+{
+    internal class PollyRetryPolicy
+    {
+        private readonly TimeSpan __one_second = TimeSpan.FromSeconds(1);
+
+        public PollyRetryPolicy(int internalRequeues)
+            => RetryPolicy =
+                Policy.HandleResult<HttpResponseMessage>(response => response.StatusCode == HttpStatusCode.TooManyRequests
+                    && response.Headers.GetValues("X-DSharpPlus-Preemptive-Ratelimit").Any())
+                    .WaitAndRetryAsync(retryCount: internalRequeues,
+                        sleepDurationProvider: (_, responseTask, _) =>
+                        {
+                            HttpResponseMessage message = responseTask.Result;
+
+                            if (message.Headers.GetValues("X-RateLimit-Scope").SingleOrDefault() == "shared")
```

Comment:
Heuristically unreachable code.

Reply:
yes, but we agreed to include this on the off chance that it ever happens, partly because the original plan was to retry requests for a bit, which would be aborted if they hit a shared ratelimit.

Reply:
Except with the way this is set up, this will *never* happen. There's no code path that allows a request to be made (in the current state of this PR).

When the bucket bug is fixed, this *may* the case, but 429s should generally be handled indiscriminately, as sub-ratelimits (i.e. scopes, excluding global) are a mess to deal with.